core/commands/up_command.rb

```diff
@@ -286,6 +310,45 @@ def generate_config_information(working_directory, config_path, node = '')
     end
   end
 
+  # Forcefully destroys given nodes
+  #
+  # @param node [Array<String>] List with names of nodes which needs to be destroyed
+  def destroy_nodes(node_names)
+    @ui.info 'Destroying existing nodes.'
+    node_names.each do |node|
+      DestroyCommand.execute(["#{@config.path}/#{node}"], @env, @ui, { keep_template: true })
+    end
+  end
+
+  # Starts shutdown nodes, restarts running nodes when up command called with --recreate option
+  #
+  # @param provider [String] name of the provider to use.
+  # @param node_names [Arrat<String>] List of nodes to start
+  # @return [Array<String>] List of nodes that were brought up
+  def start_disabled_nodes(provider, node_names)
+    running_nodes, halt_nodes = running_and_halt_nodes(node_names)
+    if @env.recreate
+      destroy_nodes(running_nodes)
+      halt_nodes = halt_nodes.concat(running_nodes)
+    end
+    halt_nodes.each do |node|
+      bring_up_machines(provider, node)
+    end
+    halt_nodes
+  end
+
+  # Restors network configuration of nodes that were already brought up
+  def restore_previous_network_config
```

Comment:
The method should be renamed into `store`. Restoration is not the word you are looking for.